lib/rules/class-methods-use-this.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @fileoverview Rule to enforce that all class methods use 'this'.
+ * @author Patrick Williams
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "enforce that class methods utilize `this`",
+            category: "Best Practices",
+            recommended: false
+        },
+        schema: []
+    },
+    create(context) {
+        const stack = [];
+
+        /**
+         * Initializes the current context to false and pushes it onto the stack.
+         * These booleans represent whether 'this' has been used in the context.
+         * @returns {void}
+         * @private
+         */
+        function enterFunction() {
+            stack.push(false);
+        }
+
+        /**
+         * Check if the node is an instance method
+         * @param {ASTNode} node - node to check
+         * @returns {boolean} True if its an instance method
+         * @private
+         */
+        function isInstanceMethod(node) {
+            return !node.static && node.kind !== "constructor" && node.type === "MethodDefinition";
+        }
+
+        /**
+         * Checks if we are leaving a function that is a method, and reports if 'this' has not been used.
+         * Static methods and the constructor are exempt.
+         * Then pops the context off the stack.
+         * @param {ASTNode} node - A function node that was entered.
+         * @returns {void}
+         * @private
+         */
+        function exitFunction(node) {
+            if (isInstanceMethod(node.parent) && !stack.pop()) {
+                context.report(node, "Expected 'this' to be used by class method '" + node.parent.key.name + "'.");
+            }
+            stack.pop();
```

Comment:
I'm guessing Travis isn't failing because there are no tests that cover nested methods. Worth adding a test for that?